### DISCUSSION TITLE
Error creating disv file from Voronoi in flopy

### DISCUSSION THREAD

**Discussion by robinkeegan** [modflowpy/flopy#2168] [General]
Hi there,

I have been trying to create an Voronoi mesh in FloPy by first generating a triangle mesh with the Triangle class/executable and then applying the Voronoi grid class to generate a Voronois. I can plot the Voronois with matplotlib but when I try and add them to a disv object I get the below error:

`An error occurred in data element "cell2d" model "model_gwf" package "disv". The error occurred while setting data in the "__init__" method. Additional Information: (1) Data line [77, 500189.2587882145, 7000754.938962412, 0] only has 4 entries, minimum number of entries is 5. (2) Error occurred while adding dataset "cell2d" to block "cell2d".`

Update: when examining cell2d it appears the cell has 0 vertices (i.e., ncvert=0). This suggests there is an issue with the geometry. 

Does anyone know what is causing this error and how I can resolve it?

Thanks,

Robin

**Comment by robinkeegan** [comment]
The error was due to the edge of a refinement polygon touching the edge of the domain polygon. The resolution involved modifying the refinement polygon to ensure no edges touched the domain polygon.

**Reply by wpbonelli** [reply]
This seems like something the Voronoi class should be able to catch and adjust for, will take a closer look.

**Reply by wpbonelli** [reply]
I'm struggling to reproduce this issue with refinements touching the edge of the domain. @robinkeegan any way you could provide a sample model?

**Reply by robinkeegan** [reply]
@wpbonelli I will email you a sample shortly.

**Reply by wpbonelli** [reply]
@robinkeegan just following up here- if you are able to send a sample model, happy to look into it. wbonelli [at] ucar [dot] edu
